### DISCUSSION TITLE
William Alligator Change fixed parameters

### DISCUSSION THREAD

**Discussion by myalgomate** [DaveSkender/Stock.Indicators#661] [Ideas]
Hi,

Can we replace the value of the fixed-parameter with the variable so that one can change the value on the arguments calculating with the indicator?
My doubt on William alligator: https://dotnet.stockindicators.dev/indicators/Alligator/#content

![image](https://user-images.githubusercontent.com/76814842/146497607-d496053c-94fa-4014-8ac7-1a828e23e1b5.png)

![image](https://user-images.githubusercontent.com/76814842/146497648-fce8e177-ed00-4ae8-abd2-04d0f70f8a40.png)

**Comment by DaveSkender** [comment]
Let me look into it.  Can you elaborate on the need or benefit to alter the original formula specs?  If I remember correctly, Mr. Williams was very specific on these values and if the indicator had other values, it doesn’t really work anymore.  I remember thinking about parameterizing these, but after reading the original author's description, and seeing how badly it doesn't work with other specs, I decided not to add those customization options.

**Reply by DaveSkender** [reply]
I was just taking a closer look at this and I think I'm okay with parameterizing these values.  It's going to be very easy to provide bad combinations of values, but I'll let users make that determination on their own.  With that said, I'll try to add some validations to make sure they are provided sensically; for example, the Jaws period should never be smaller than the Teeth period.

**Reply by DaveSkender** [reply]
You can follow #662 for tracking purposes.

**Reply by myalgomate** [reply]
Thanks. Hope it would be useful to others as well.

**Reply by DaveSkender** [reply]
This parameterization is now available in v1.20.1

**Reply by myalgomate** [reply]
Thanks
